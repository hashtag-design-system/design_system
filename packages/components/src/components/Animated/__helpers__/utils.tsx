import { screen } from "@testing-library/react";

export const checkChildStyle = (element: string | HTMLElement, styleValue: { name: string; value: React.ReactText }) => {
  const { name, value } = styleValue;
  if (typeof element === "string") {
    expect(screen.getByTestId(element).style["_values"][`--${name}`]).toBe(String(value));
  } else {
    expect(element.style["_values"][`--${name}`]).toBe(String(value));
  }
};

export const checkStyleCustomProperties = (element: HTMLElement, arr: string[]) => {
  arr
    .map(item => `--${item}`)
    .forEach((style, i) => {
      expect(element.style[i.toString()]).toBe(style);
    });
};
