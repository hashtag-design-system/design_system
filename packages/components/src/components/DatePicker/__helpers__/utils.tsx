import dayjs from "dayjs";
import React from "react";
import Select from "../../Select";
import DatePicker, { DatePickerFProps, DatePickerOnChangeInfo } from "../index";

type FProps = DatePickerFProps;

export const TEST_DEFAULT_DATE = dayjs("2021-01-26");

export const DefaultDatePicker: React.FC<FProps> = ({ ...props }) => {
  return <DatePicker defaultOpen defaultDates={[TEST_DEFAULT_DATE]} defaultCalendarDate={TEST_DEFAULT_DATE} {...props} />;
};

export const TestDatePickerButton: React.FC<Pick<DatePickerOnChangeInfo, "selectedDate">> = ({ selectedDate }) => (
  <Select.Button>{selectedDate.length >= 1 ? selectedDate[0].format("DD/MM/YYYY") : ""}</Select.Button>
);
