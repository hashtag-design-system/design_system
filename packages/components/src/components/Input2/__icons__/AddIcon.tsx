import React from "react";

export const AddIcon: React.FunctionComponent = () => {
  return (
    <svg width={24} height={24} viewBox="0 0 24 24" className="icon" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path d="M3.429 12H20.57M12 20.572V3.429" stroke="#0303FF" strokeWidth={2.286} strokeLinecap="round" />
    </svg>
  );
};
