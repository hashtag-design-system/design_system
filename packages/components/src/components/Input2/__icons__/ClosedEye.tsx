import React from "react";

export const ClosedEye: React.FunctionComponent = () => {
  return (
    <svg width={24} height={24} viewBox="0 0 24 24" fill="none" className="icon" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M8.678 6.924A8.327 8.327 0 0112 6.235c2.487 0 4.468 1.083 5.942 2.36a13.815 13.815 0 012.783 3.36c.01.016.013.031.013.044a.1.1 0 01-.015.05 13.795 13.795 0 01-1.288 1.826.609.609 0 00.092.869.64.64 0 00.888-.09 15.024 15.024 0 001.404-1.993 1.3 1.3 0 00.002-1.319 15.051 15.051 0 00-3.043-3.673C17.154 6.263 14.888 5 12 5a9.613 9.613 0 00-3.83.793.613.613 0 00-.324.814c.14.313.513.454.832.317z"
        fill="#000"
      />
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M18.03 16.93C16.485 18.078 14.469 19 12 19c-2.888 0-5.154-1.263-6.778-2.669a15.054 15.054 0 01-3.043-3.673c-.24-.413-.237-.91.001-1.318A15.12 15.12 0 014.972 7.89L3.01 6.534a.61.61 0 01-.15-.861.64.64 0 01.879-.148L20.99 17.466a.61.61 0 01.151.861.64.64 0 01-.88.148l-2.23-1.544zM6.028 8.623a13.826 13.826 0 00-2.75 3.33.096.096 0 00-.015.049c0 .012.003.027.014.045a13.815 13.815 0 002.783 3.36C7.532 16.68 9.513 17.765 12 17.765c1.964 0 3.61-.675 4.943-1.587l-2.959-2.048a2.97 2.97 0 01-1.984.752c-1.627 0-2.945-1.29-2.945-2.882 0-.398.082-.777.231-1.122l-3.26-2.256z"
        fill="#000"
      />
      <path
        d="M14.89 11.882c0 1.592-1.318 2.883-2.945 2.883-1.626 0-2.945-1.29-2.945-2.883C9 10.291 10.319 9 11.945 9c1.627 0 2.945 1.29 2.945 2.882z"
        fill="#000"
      />
    </svg>
  );
};
