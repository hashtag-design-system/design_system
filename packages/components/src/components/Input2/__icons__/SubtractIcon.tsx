import React from "react";

export const SubtractIcon: React.FunctionComponent = () => {
  return (
    <svg width={24} height={24} viewBox="0 0 24 24" fill="none" className="icon" xmlns="http://www.w3.org/2000/svg">
      <path d="M3.429 12.953H20.57" stroke="#999" strokeWidth={2.286} strokeLinecap="round" />
    </svg>
  );
};
