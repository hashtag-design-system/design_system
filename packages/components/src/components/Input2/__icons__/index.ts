export * from "./AddIcon";
export * from "./ClosedEye";
export * from "./OpenEye";
export * from "./SubtractIcon";
