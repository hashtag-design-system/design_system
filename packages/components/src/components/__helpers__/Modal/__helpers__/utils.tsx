import { screen, waitFor } from "@testing-library/react";
import { CONFIG } from "../../../../config";

export const overlayCheckStyle = async (toContain: string): Promise<void> => {
  const modal = screen.getByTestId("modal");

  // Wait due to animation from `opacity: 0` to `opacity: 1`
  await waitFor(() => {
    expect(modal).toBeVisible();
  });

  expect(modal).toHaveAttribute("style");
  if (toContain.includes("rgba")) {
    expect(modal.style.backgroundColor).toContain(toContain);
  } else {
    expect(modal.style["backdropFilter"]).toContain(toContain);
  }
};

export const createPortalElement = () => {
  const newEl = document.createElement("div");
  newEl.setAttribute("id", CONFIG.PORTAL_ID_SELECTOR);
  newEl.setAttribute("data-testid", CONFIG.PORTAL_ID_SELECTOR);
  document.body.prepend(newEl);
  return newEl;
};