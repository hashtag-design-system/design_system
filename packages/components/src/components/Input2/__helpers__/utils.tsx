import { render, RenderOptions } from "@testing-library/react";
import React from "react";
import { InputContextProvider, generateInputId } from "../../../utils";
import { InputFProps } from "../index";

export const defaultProps: InputFProps = {
  id: generateInputId(),
  placeholder: "Placeholder",
  floatingplaceholder: true,
  state: "default",
};

type InputCustomRenderOptions = {
  providerProps: InputFProps;
  renderOptions?: RenderOptions;
};

export const inputCustomRender = (ui: React.ReactElement, options?: InputCustomRenderOptions) => {
  return render(
    <InputContextProvider value={{ ...defaultProps, ...options?.providerProps }}>{ui}</InputContextProvider>,
    options?.renderOptions
  );
};
