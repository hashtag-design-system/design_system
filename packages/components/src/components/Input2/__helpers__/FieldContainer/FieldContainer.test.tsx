import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { ClosedEye } from "../../__icons__/ClosedEye";
import { OpenEye } from "../../__icons__/OpenEye";
import { inputCustomRender } from "../utils";
import { FieldContainer } from "./FieldContainer";

describe("Input <FieldContainer />", () => {
  test("characterLimit & maxLength", async () => {
    const maxLength = 14;
    inputCustomRender(<FieldContainer />, { providerProps: { characterLimit: true, maxLength } });
    const input = screen.getByTestId("input");
    const newVal = "test123@gmail.com";

    expect(input).toBeInTheDocument();

    userEvent.type(input, newVal);
    expect(input).toHaveValue(newVal.substring(0, maxLength));
    expect(input.getAttribute("value")).toHaveLength(maxLength);
  });
  describe("prefix & suffix of string type", () => {
    test("displays prefix", () => {
      inputCustomRender(<FieldContainer />, { providerProps: { prefix: "$" } });

      expect(screen.getByTestId("prefix")).toBeInTheDocument();
    });
    test("displays suffix", () => {
      inputCustomRender(<FieldContainer />, { providerProps: { suffix: "Kg" } });

      expect(screen.getByTestId("suffix")).toBeInTheDocument();
    });
    test("displays both prefix & suffix", () => {
      inputCustomRender(<FieldContainer />, { providerProps: { prefix: "$", suffix: "Kg" } });

      expect(screen.getByTestId("suffix")).toBeInTheDocument();
      expect(screen.getByTestId("prefix")).toBeInTheDocument();
    });
  });
  describe("prefix & suffix of `React.ReactNode` (component - icon) type", () => {
    test("displays prefix component", () => {
      inputCustomRender(<FieldContainer />, { providerProps: { prefix: <OpenEye /> } });
      const prefix = screen.getByTestId("prefix");

      expect(prefix).toBeInTheDocument();
      expect(prefix).toMatchSnapshot();
    });
    test("displays suffix component", () => {
      inputCustomRender(<FieldContainer />, { providerProps: { suffix: <OpenEye /> } });
      const suffix = screen.getByTestId("suffix");

      expect(suffix).toBeInTheDocument();
      expect(suffix).toMatchSnapshot();
    });
    test("displays both prefix & suffix components", () => {
      inputCustomRender(<FieldContainer />, { providerProps: { prefix: <OpenEye />, suffix: <ClosedEye /> } });

      const suffix = screen.getByTestId("suffix");
      const prefix = screen.getByTestId("prefix");

      expect(suffix).toBeInTheDocument();
      expect(suffix).toMatchSnapshot();
      expect(prefix).toBeInTheDocument();
      expect(prefix).toMatchSnapshot();
    });
  });
  describe("<Base /> style", () => {
    test("with prefix", () => {
      inputCustomRender(<FieldContainer />, { providerProps: { prefix: "Kg" } });
      const input = screen.getByTestId("input");
      const paddingLeft = input.style.paddingLeft;

      expect(paddingLeft).toBeDefined();
      expect(typeof paddingLeft).toBe("string");
      expect(paddingLeft).toContain("em");
    });
    test("with suffix", () => {
      inputCustomRender(<FieldContainer />, { providerProps: { suffix: "Kg" } });
      const input = screen.getByTestId("input");
      const paddingRight = input.style.paddingRight;

      expect(paddingRight).toBeDefined();
      expect(typeof paddingRight).toBe("string");
      expect(paddingRight).toContain("em");
    });
  });
  // test("with characterLimit", async () => {
  //     const maxLength = 14;
  //     inputCustomRender(<FieldContainer />, { providerProps: { characterLimit: true, maxLength } });
  //     const input = screen.getByTestId("input");
  //     const newVal = "test123@gmail.com";
  //     const characterLimit = screen.getByTestId("character-limit");

  //     expect(screen.getByTestId("second-help-text-container")).toBeInTheDocument();
  //     expect(input).toBeInTheDocument();
  //     expect(characterLimit).toBeInTheDocument();

  //     await waitFor(() => {
  //       userEvent.type(input, newVal);
  //     });
  //     expect(input).toHaveValue(newVal.substring(0, maxLength));
  //     expect(input.getAttribute("value")).toHaveLength(maxLength);
  //     expect(characterLimit).toHaveTextContent(`${maxLength} / ${maxLength}`);
  //     expect(characterLimit).toHaveClass("medium");
  //   });
});
