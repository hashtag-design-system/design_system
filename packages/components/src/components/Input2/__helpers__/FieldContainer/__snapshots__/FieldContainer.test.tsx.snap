// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input <FieldContainer /> prefix & suffix of \`React.ReactNode\` (component - icon) type displays both prefix & suffix components 1`] = `
<div
  class="input__icon right"
  data-testid="suffix"
  style="transform: translateY(-40%);"
>
  <svg
    class="icon"
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.678 6.924A8.327 8.327 0 0112 6.235c2.487 0 4.468 1.083 5.942 2.36a13.815 13.815 0 012.783 3.36c.01.016.013.031.013.044a.1.1 0 01-.015.05 13.795 13.795 0 01-1.288 1.826.609.609 0 00.092.869.64.64 0 00.888-.09 15.024 15.024 0 001.404-1.993 1.3 1.3 0 00.002-1.319 15.051 15.051 0 00-3.043-3.673C17.154 6.263 14.888 5 12 5a9.613 9.613 0 00-3.83.793.613.613 0 00-.324.814c.14.313.513.454.832.317z"
      fill="#000"
    />
    <path
      clip-rule="evenodd"
      d="M18.03 16.93C16.485 18.078 14.469 19 12 19c-2.888 0-5.154-1.263-6.778-2.669a15.054 15.054 0 01-3.043-3.673c-.24-.413-.237-.91.001-1.318A15.12 15.12 0 014.972 7.89L3.01 6.534a.61.61 0 01-.15-.861.64.64 0 01.879-.148L20.99 17.466a.61.61 0 01.151.861.64.64 0 01-.88.148l-2.23-1.544zM6.028 8.623a13.826 13.826 0 00-2.75 3.33.096.096 0 00-.015.049c0 .012.003.027.014.045a13.815 13.815 0 002.783 3.36C7.532 16.68 9.513 17.765 12 17.765c1.964 0 3.61-.675 4.943-1.587l-2.959-2.048a2.97 2.97 0 01-1.984.752c-1.627 0-2.945-1.29-2.945-2.882 0-.398.082-.777.231-1.122l-3.26-2.256z"
      fill="#000"
      fill-rule="evenodd"
    />
    <path
      d="M14.89 11.882c0 1.592-1.318 2.883-2.945 2.883-1.626 0-2.945-1.29-2.945-2.883C9 10.291 10.319 9 11.945 9c1.627 0 2.945 1.29 2.945 2.882z"
      fill="#000"
    />
  </svg>
</div>
`;

exports[`Input <FieldContainer /> prefix & suffix of \`React.ReactNode\` (component - icon) type displays both prefix & suffix components 2`] = `
<div
  class="input__icon left"
  data-testid="prefix"
  style="transform: translateY(-40%);"
>
  <svg
    class="icon"
    data-testid="icon"
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14.945 12c0 1.592-1.319 2.882-2.945 2.882-1.627 0-2.945-1.29-2.945-2.882S10.373 9.118 12 9.118c1.627 0 2.945 1.29 2.945 2.882z"
    />
    <path
      clip-rule="evenodd"
      d="M12 5C9.112 5 6.846 6.263 5.223 7.669a15.052 15.052 0 00-3.043 3.673c-.24.411-.24.905 0 1.316.408.7 1.422 2.27 3.043 3.673C6.846 17.737 9.113 19 12 19c2.888 0 5.154-1.264 6.777-2.669a15.05 15.05 0 003.043-3.673c.24-.411.24-.905 0-1.316a15.052 15.052 0 00-3.043-3.673C17.154 6.263 14.887 5 12 5zm-8.724 6.954a13.815 13.815 0 012.783-3.36C7.533 7.318 9.513 6.235 12 6.235c2.487 0 4.467 1.083 5.941 2.36a13.812 13.812 0 012.783 3.36.09.09 0 01.014.045.09.09 0 01-.014.046 13.813 13.813 0 01-2.783 3.36c-1.474 1.275-3.454 2.359-5.941 2.359-2.487 0-4.467-1.084-5.941-2.36a13.814 13.814 0 01-2.783-3.36.09.09 0 01-.014-.045.09.09 0 01.014-.046z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`Input <FieldContainer /> prefix & suffix of \`React.ReactNode\` (component - icon) type displays prefix component 1`] = `
<div
  class="input__icon left"
  data-testid="prefix"
  style="transform: translateY(-40%);"
>
  <svg
    class="icon"
    data-testid="icon"
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14.945 12c0 1.592-1.319 2.882-2.945 2.882-1.627 0-2.945-1.29-2.945-2.882S10.373 9.118 12 9.118c1.627 0 2.945 1.29 2.945 2.882z"
    />
    <path
      clip-rule="evenodd"
      d="M12 5C9.112 5 6.846 6.263 5.223 7.669a15.052 15.052 0 00-3.043 3.673c-.24.411-.24.905 0 1.316.408.7 1.422 2.27 3.043 3.673C6.846 17.737 9.113 19 12 19c2.888 0 5.154-1.264 6.777-2.669a15.05 15.05 0 003.043-3.673c.24-.411.24-.905 0-1.316a15.052 15.052 0 00-3.043-3.673C17.154 6.263 14.887 5 12 5zm-8.724 6.954a13.815 13.815 0 012.783-3.36C7.533 7.318 9.513 6.235 12 6.235c2.487 0 4.467 1.083 5.941 2.36a13.812 13.812 0 012.783 3.36.09.09 0 01.014.045.09.09 0 01-.014.046 13.813 13.813 0 01-2.783 3.36c-1.474 1.275-3.454 2.359-5.941 2.359-2.487 0-4.467-1.084-5.941-2.36a13.814 13.814 0 01-2.783-3.36.09.09 0 01-.014-.045.09.09 0 01.014-.046z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`Input <FieldContainer /> prefix & suffix of \`React.ReactNode\` (component - icon) type displays suffix component 1`] = `
<div
  class="input__icon right"
  data-testid="suffix"
  style="transform: translateY(-40%);"
>
  <svg
    class="icon"
    data-testid="icon"
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14.945 12c0 1.592-1.319 2.882-2.945 2.882-1.627 0-2.945-1.29-2.945-2.882S10.373 9.118 12 9.118c1.627 0 2.945 1.29 2.945 2.882z"
    />
    <path
      clip-rule="evenodd"
      d="M12 5C9.112 5 6.846 6.263 5.223 7.669a15.052 15.052 0 00-3.043 3.673c-.24.411-.24.905 0 1.316.408.7 1.422 2.27 3.043 3.673C6.846 17.737 9.113 19 12 19c2.888 0 5.154-1.264 6.777-2.669a15.05 15.05 0 003.043-3.673c.24-.411.24-.905 0-1.316a15.052 15.052 0 00-3.043-3.673C17.154 6.263 14.887 5 12 5zm-8.724 6.954a13.815 13.815 0 012.783-3.36C7.533 7.318 9.513 6.235 12 6.235c2.487 0 4.467 1.083 5.941 2.36a13.812 13.812 0 012.783 3.36.09.09 0 01.014.045.09.09 0 01-.014.046 13.813 13.813 0 01-2.783 3.36c-1.474 1.275-3.454 2.359-5.941 2.359-2.487 0-4.467-1.084-5.941-2.36a13.814 13.814 0 01-2.783-3.36.09.09 0 01-.014-.045.09.09 0 01.014-.046z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;
